rosbot_gazebo/test/simulation_test_node.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+# Copyright 2023 Husarion
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import rclpy
+
+from threading import Event
+from threading import Thread
+
+from rclpy.node import Node
+
+from geometry_msgs.msg import Twist
+from nav_msgs.msg import Odometry
+import tf_transformations
+
+
+class SimulationTestNode(Node):
+    __test__ = False
+
+    def __init__(self, name="test_node"):
+        super().__init__(name)
+        self.goal_x_distance = 0.0
+        self.goal_y_distance = 0.0
+        self.goal_theta_angle = 0.0
+
+        self.goal_x_event = Event()
+        self.goal_y_event = Event()
+        self.goal_theta_event = Event()
+
+    def set_and_publish_velocities(
+        self, goal_x_distance, goal_y_distance, goal_theta_angle
+    ):
+        self.goal_x_distance = goal_x_distance
+        self.goal_y_distance = goal_y_distance
+        self.goal_theta_angle = goal_theta_angle
+        self.publish_cmd_vel_messages()
```

Comment:
It would be useful to establish a name. The name of the function suggests speed and the attributes are called distance.